xl_auth/token/models.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+"""Token models."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from codecs import getencoder
+from datetime import datetime, timedelta
+from os import urandom
+
+from ..database import Column, Model, SurrogatePK, db, relationship
+
+
+class Token(SurrogatePK, Model):
+    """An OAuth2 Bearer token."""
+
+    __tablename__ = 'tokens'
+    user_id = Column(db.Integer, db.ForeignKey('users.id', ondelete='CASCADE'), nullable=False)
+    user = relationship('User')
+
+    client_id = Column(db.String(64), db.ForeignKey('clients.client_id', ondelete='CASCADE'),
+                       nullable=False)
+    client = relationship('Client')
+
+    token_type = Column(db.String(40), nullable=False)
```

Comment:
Hmm.. Bad idea maybe. Better to declare the default value using the built-in kwarg instead of putting it in `__init__()`, methinks. I.e.

```python
    token_type = Column(db.String(40), nullable=False, default='bearer')
```

Reply:
Assigning in `__init__()`, since it's theoretically possible that we will want to change this in the future.